src/charm_state.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""Module that handles Pollen charm's state."""
+
+from pydantic import BaseModel, Extra
+
+HTTP_PORT = "443"
+
+
+class WebsiteModel(BaseModel, extra=Extra.forbid):
+    """Pydantic validation model for the website property.
+
+    Attrs:
+        hostname: Website hostname.
+        port: Website port.
+    """
+
+    hostname: str
+    port: str
+
+
+class CharmState:
+    """Class that represents the state of the Pollen charm.
+
+    Attrs:
+        website: Website to pass over the website relation
+    """
+
+    def __init__(self, hostname: str) -> None:
+        """Construct.
+
+        Args:
+            hostname: hostname to build the website property
+        """
+        self._hostname = hostname
+        self.rng_tools_file = None
```

Comment:
I can't find where the content of this attribute is set based on the file content. There's a method that dumps the content in the file if this attribute is empty, but nowhere the file is read to get its content